app/controllers/warehouses_controller.rb

```diff
@@ -0,0 +1,75 @@
+class WarehousesController < ApplicationController
+  before_action :authenticate_admin!
+  before_action :set_warehouse, only: [:show, :edit, :update, :destroy]
```

Comment:
in the ruby world, we tend no to use 'getters' and 'setters' instead of get_something use something like find_warehouse, required_data or something